### PR TITLE
DB-11957 Query hinting mechanism

### PR DESCRIPTION
**Pull request by dgomezferro**
## Short Description
Add query hinting mechanism

### CODE REVIEW

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2848/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by arnaud-splice — APPROVED** [review]

**Review by jyuanca — APPROVED** [review]
License header is missing for some new files

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1 on content, but do not merge to Master until details on PR 5416 and PR 5429 are worked out.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/QueryHintingService.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ */
+
+package com.splicemachine.db.impl.sql.compile;
+
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class QueryHintingService {
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/QueryHintingService.java:22]: ahhh cool

**Pull request closed without merging**